src/rard/research/models/concordance.py

```diff
@@ -0,0 +1,102 @@
+from django.db import models
+from django.db.models import Case, IntegerField, When
+
+from rard.research.models.mixins import HistoryModelMixin
+from rard.utils.basemodel import BaseModel
+
+
+class PartIdentifier(HistoryModelMixin, BaseModel):
+    class Meta:
+        ordering = ["edition__name", "value"]
+
+    def __str__(self):
+        if self.is_template:
+            return f"format for {self.edition.name}: {self.value}"
+        else:
+            return f"{self.edition.name}: {self.value}"
+
+    edition = models.ForeignKey("Edition", on_delete=models.CASCADE)
+    value = models.CharField(max_length=250, blank=False)
+    display_order = models.CharField(max_length=30, blank=True)
+
+    @property
+    def is_template(self):
+        return "[" in self.value and "]" in self.value
+
+
+class Edition(HistoryModelMixin, BaseModel):
+    def __str__(self, bib=False):
+        if bib:
+            return self.description
+        else:
+            return f"{self.name} ({self.description})"
+
+    name = models.CharField(max_length=128, blank=False)
+
+    description = models.CharField(
+        max_length=200, blank=True, null=True
+    )  # need to add this to master bib
+
+
+class ConcordanceModel(HistoryModelMixin, BaseModel):
```

Comment:
I can't see a display_order field here, but it's on the form. What happens to that?

Reply:
it's attached to the `PartIdentifier` model since that's what it applies to